apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/SessionPoolTests.TargetedSessionPoolTests.cs

```diff
@@ -333,6 +333,40 @@ public async Task AcquireAsync_MaxActiveSessions_Block()
                 });
             }
 
+            [Fact(Timeout = TestTimeoutMilliseconds)]
+            public async Task AcquireDetachAsync()
+            {
+                var pool = CreatePool(false);
+                var client = (SessionTestingSpannerClient) pool.Client;
+                pool.Options.WaitOnResourcesExhausted = ResourcesExhaustedBehavior.Fail;
+
+                await client.Scheduler.RunAsync(async () =>
+                {
+                    // We acquire all sessions to make certain we don't get new ones.
+                    var sessions = await AcquireAllSessionsAsync(pool);
+                    // Sanity check, maximum possible of active sessions are active.
+                    Assert.Equal(pool.Options.MaximumActiveSessions, pool.ActiveSessionCount);
+                    Assert.Equal(pool.Options.MaximumActiveSessions, pool.LiveOrRequestedSessionCount);
+
+                    // Now let's return one session to the pool so that we can acquire that one, but detached.
+                    var firstSession = sessions[0];
+                    firstSession.ReleaseToPool(false);
+                    // Sanity check, there's one less active session than possible, we just returned one to the pool.
+                    Assert.Equal(pool.Options.MaximumActiveSessions - 1, pool.ActiveSessionCount);
+                    // But the pool is still tracking MaximumActiveSessions
+                    Assert.Equal(pool.Options.MaximumActiveSessions, pool.LiveOrRequestedSessionCount);
+
+                    var detachedSession = await pool.AcquireSessionAsync(new TransactionOptions(), singleUseTransaction : false, detached: true, default);
+                    // Even though we just acquired a session, we acquired it detached so the pool stopped tracking this session.
+                    Assert.Equal(pool.Options.MaximumActiveSessions - 1, pool.ActiveSessionCount);
+                    Assert.Equal(pool.Options.MaximumActiveSessions - 1, pool.LiveOrRequestedSessionCount);
+
+                    // Even though we configured the pool to fail on resource exahsusted, acquiring a new session won't fail.
+                    // Resources are not exhausted because we acquired one of the sessions as detached.
+                    var sessionAcquitistionTask = pool.AcquireSessionAsync(new TransactionOptions(), singleUseTransaction: false, detached: false, default);
```

Comment:
Nit: sessionAcquitistionTask => sessionAcquisitionTask

And shouldn't we await that task, to validate that it succeeds?

Reply:
Yes on both.

Althouhg on resource exhausted it fails inmediately, i.e. before becoming async. Still, I'll modify the test in case that behaviour changes in the future.